app/questionnaire/questionnaire_store_updater.py

```diff
@@ -131,12 +131,14 @@ def add_primary_person(self, list_name: str) -> str:
         # If a primary person was initially answered negatively, then changed to positive,
         # the location must be removed from the progress store.
         self.remove_completed_location(self._current_location)
+        self._capture_dependencies_for_list_change(list_name)
 
         return self._list_store.add_list_item(list_name, primary_person=True)
 
     def add_list_item(self, list_name: str) -> str:
         new_list_item_id = self._list_store.add_list_item(list_name)
         self.remove_completed_relationship_locations_for_list_name(list_name)
+        self._capture_dependencies_for_list_change(list_name)
```

Comment:
This would be better done at the source, i.e handler, or SDS. Being here means the `_capture_dependencies_for_list_change` is called multiple times when SDS is adding list items, or removing list items, even if they are all for the same list name.

Reply:
Yeah, I wanted to minimise what callers need to do, but agree this probably does get called too much if done like that - made it public and moved to callers 👍